src/coverlet.core/Coverage.cs

```diff
@@ -170,11 +183,22 @@ public CoverageResult GetCoverageResult()
                 InstrumentationHelper.RestoreOriginalModule(result.ModulePath, _identifier);
             }
 
-            var coverageResult = new CoverageResult { Identifier = _identifier, Modules = modules, InstrumentedResults = _results };
+            var coverageResult = new CoverageResult
+                {Identifier = _identifier, Modules = modules, InstrumentedResults = _results};
 
             if (!string.IsNullOrEmpty(_mergeWith) && !string.IsNullOrWhiteSpace(_mergeWith) && File.Exists(_mergeWith))
             {
-                string json = File.ReadAllText(_mergeWith);
+                var json = RetryHelper.Do(() =>
```

Comment:
Please can you add some comment above like `// Possible concurrency access to merge file between write/read on parallel testing`  here and below?